weave.tf

```diff
@@ -118,12 +118,62 @@ resource "kubernetes_deployment" "weave" {
               memory = "16G"
             }
           }
+
+          dynamic "volume_mount" {
+            for_each = var.weave_storage_size != "" ? ["weave-cache"] : []
+            content {
+            name       = volume_mount.value
+              mount_path = "/vol/weave/cache"
+            }
+          }
+        }
+        dynamic "volume" {
+          for_each = var.weave_storage_size != "" ? ["weave-cache"] : []
```

Comment:
In reference to my previous comment, using a number type and noting the requirement to do so ("size in megabytes/gigabytes", etc.) will avoid situations like users providing "100Mib", "100Mb", "100" -- each of which is syntactically valid, but may not work, particularly if these values are being passed through to as a template to the helm provider.